src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -976,46 +975,43 @@ void StorageReplicatedMergeTree::setTableStructure(
     }
 
     /// Changes in columns may affect following metadata fields
-    if (new_metadata.columns != old_metadata.columns)
+    new_metadata.column_ttls_by_name.clear();
```

Comment:
So `ColumnDescription` comparison operator works incorrectly for `Decimal` types, because `DataTypeDecimal<T>::equals(...)` does not compare precision, right? Can we fix `DataTypeDecimal<T>::equals(...)` instead of `setTableStructure(...)`?

Reply:
But I'm not sure of the consequences :) The fix is trivial ([link](https://github.com/ClickHouse/ClickHouse/blob/jepsen_for_nukeeper/src/DataTypes/DataTypesDecimal.cpp#L37-L38)), but we can get something like additional casts where types are actually equal under the hood.